engine/classes/Elgg/Di/DiContainer.php

```diff
@@ -47,18 +42,15 @@ class DiContainer {
 	 * @throws \Elgg\Di\MissingValueException
 	 */
 	public function __get($name) {
-		if (array_key_exists($name, $this->cache)) {
-			return $this->cache[$name];
-		}
-		if (!isset($this->factories[$name])) {
+		if (!isset($this->__factories[$name])) {
 			throw new \Elgg\Di\MissingValueException("Value or factory was not set for: $name");
 		}
-		$value = $this->build($this->factories[$name]['callable'], $name);
+		$value = $this->build($this->__factories[$name]['callable'], $name);
```

Comment:
I figured dependencies beginning with a single underscore might be of some use. Maybe not.